protocol-tests/swagger/enums.json

```diff
@@ -0,0 +1,175 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Enum Service",
+    "description": "Enum Service",
+    "version": "1.0.0"
+  },
+  "host": "localhost:3000",
+  "schemes": [
+    "http"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "paths": {
+    "/enums/queryparam": {
+      "put": {
+        "operationId": "putQueryEnums",
+        "description": "query enums operation",
+        "parameters": [
+          {
+            "name": "query-integer-enum",
+            "in": "query",
+            "required": true,
+            "type": "integer",
+            "description": "integer enum with three values",
+            "enum": [
+              100,
+              200,
+              300
+            ],
+            "x-ms-enum": {
+              "modelAsString": false
+            }
+          },
+          {
+            "name": "query-boolean-enum",
+            "in": "query",
+            "required": true,
+            "type": "boolean",
+            "description": "boolean enum with two values",
+            "enum": [
+              true,
+              false
+            ],
+            "x-ms-enum": {
+              "modelAsString": false
+            }
+          },
+          {
+            "name": "query-non-required-enum",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "non required enum with three values",
+            "enum": [
+              "test1",
+              "test2",
+              "test@</spec.,i`~!&*-al@char/>"
```

Comment:
in some corner case, "@abc" could fail javadoc, but let's ignore it for now.

Reply:
Javadoc looks like this:
![image](https://user-images.githubusercontent.com/87355844/173776705-f9b38a16-0acf-4b1b-ae9a-1149d35f4415.png)

Reply:
And put this to a required param as well.